src/sentry/static/sentry/app/views/integrationPipeline/awsLambdaCloudformation.tsx

```diff
@@ -1,30 +1,79 @@
 import React from 'react';
 import styled from '@emotion/styled';
+import * as qs from 'query-string';
 
 import {addErrorMessage, addLoadingMessage} from 'app/actionCreators/indicator';
-import Link from 'app/components/links/link';
+import ExternalLink from 'app/components/links/externalLink';
 import {t, tct} from 'app/locale';
+import {uniqueId} from 'app/utils/guid';
 import Form from 'app/views/settings/components/forms/form';
 import JsonForm from 'app/views/settings/components/forms/jsonForm';
 import FormModel from 'app/views/settings/components/forms/model';
 import {JsonFormObject} from 'app/views/settings/components/forms/type';
 
+// let the browser generate and store the external ID
+// this way the same user always has the same external ID if they restart the pipeline
+const ID_NAME = 'AWS_EXTERNAL_ID';
+const getAwsExternalId = () => {
+  let awsExternalId = window.localStorage.getItem(ID_NAME);
+  if (!awsExternalId) {
+    awsExternalId = uniqueId();
+    window.localStorage.setItem(ID_NAME, awsExternalId);
+  }
+  return awsExternalId;
+};
+
 type Props = {
-  cloudformationUrl: string;
-  awsExternalId: string;
+  baseCloudformationUrl: string;
+  templateUrl: string;
+  stackName: string;
+  arn?: string;
+  error?: string;
 };
 
 export default class AwsLambdaCloudformation extends React.Component<Props> {
-  model = new FormModel({apiOptions: {baseUrl: window.location.origin}});
+  componentDidMount() {
+    // show the error if we have it
+    const {error} = this.props;
+    if (error) {
+      addErrorMessage(error, {duration: 10000});
```

Comment:
@scefali I think we should add the translation `tct` for the error message

Reply:
@NisanthanNanthakumar You can't actually do translations with dynamic text like that but I can put the translations on the backend for the error message string.